Keas.Mvc/ClientApp/components/Keys/AssociateSpace.tsx

```diff
@@ -147,91 +94,74 @@ export default class AssociateSpace extends React.Component<IProps, IState> {
     return (
       <KeyEditValues
         selectedKey={selectedKeyInfo.key}
-        changeProperty={this._changeProperty}
+        changeProperty={changeProperty}
         disableEditing={false}
         searchableTags={searchableTags}
       />
     );
-  }
-
-  private _changeProperty = (property: string, value: string) => {
-    this.setState(
-      prevState => ({
-        selectedKeyInfo: {
-          ...prevState.selectedKeyInfo,
-          key: {
-            ...prevState.selectedKeyInfo.key,
-            [property]: value
-          }
-        }
-      }),
-      this._validateState
-    );
+  };
+
+  const changeProperty = (property: string, value: string) => {
+    setSelectedKeyInfo({ ...selectedKeyInfo, [property]: value });
```

Comment:
Yes. I thought in the `changeProperty` function it was just editing one property of the KeyInfo object and updated the state in it. Should it have been more like this:
```
let newKeyInfo = KeyInfo;
newKeyInfo[property] = value
setSelectedInfo(newKeyInfo)
```
?

Reply:
No because the property to change i think is on the key and not the KeyInfo object.  I think it needs to be like this:
```
setSelectedKeyInfo({ ...selectedKeyInfo, key: {
            ...selectedKeyInfo.key,
            [property]: value
          } });
```

Reply:
ooohhhh okay I see where I messed up. I will fix this asap